packages/image_picker/image_picker_for_web/lib/image_picker_for_web.dart

```diff
@@ -157,8 +163,11 @@ class ImagePickerPlugin extends ImagePickerPlatform {
     double? maxWidth,
     double? maxHeight,
     int? imageQuality,
-  }) {
-    return getFiles(accept: _kAcceptImageMimeType, multiple: true);
+  }) async {
+    return await Future.wait(
+        (await getFiles(accept: _kAcceptImageMimeType, multiple: true)).map(
+            (e) => _imageResizer.resizeImageIfNeeded(
+                e, maxWidth, maxHeight, imageQuality)));
```

Comment:
Do you need the top-level await Future.wait? Isn't this enough?

```dart
return (await getFiles(...)).map( (e) => _imageResizer.resize(...) );
```

Reply:
_imageResizer.resize() returns a future. so if we use this then return type is `Future<List<Future<Xfile>>>`  but we need `Future<List<XFIile>> so we need the future.wait function.